### DISCUSSION TITLE
Rework of the project structure. Using FallbackResource instead of Rewriting.

### DISCUSSION THREAD

**Discussion by ChazyTheBest** [PrestaShop/PrestaShop#33403] [Ideas]
#### Is your feature request related to a problem?

Yes:

1. The use of .htaccess and mod_rewrite is not efficient at all, so instead I simply use FallbackResource. If I enable SEO/friendly URLs some images in the html won't have a correct URL (due to .htaccess files and mod_rewrite being disabled).

2. The current structure is a mess, mainly because most of the files are not needed inside the document root and this involves the use of hacks such as .htaccess (which slows down the load times) and index.php files everywhere. Also, you can't have the project files outside of the document root without breaking the whole application (most likely just because of point 3).

3. Asset/static files are all scattered around, I gave up trying to find them all. If I don't have a CDN but still want to serve these files from a different domain/subdomain it's almost impossible.

#### Describe the solution you'd like

Rework the file structure to allow the following:

1. Replace mod_rewrite with mod_dir's FallbackResource (which is available since Apache 2.2.16). It can still be used inside a .htaccess file if needed/wanted. This will greatly simplify the current Apache configuration.

2. Possibility to have all server-side files outside of the document root (except the entry script). For example, having all public files inside a folder in the project root: `/Prestashop/public_html`, which would be the path used for the virtual host.

3. Possibility to have all asset/static files in the same path tree, so they can be moved to a different folder/subdomain. For example, the way Yii2 framework handles it (I'm not aware if Symfony already has a similar feature), which in this example it'd be: `/Prestashop/public_html/assets`, and then two configurations: `basePath` and `baseUrl`.

#### Additional context

The goal of this request is to have a simpler, more organized structure that doesn't require hacks, and therefore, gain a performance boost as a result.

[a comment by an automated account is not shown]

**Comment by florine2623** [comment]
Hi @ChazyTheBest ,
Thanks for the suggestion, I will ping our teams so they can decide what to do about it !

Please stay tuned.

**Comment by florine2623** [comment]
Hi @PrestaShop/prestashop-core-developers @PrestaShop/prestashop-product-team ,

Do you think it would be a possibility to rework on PrestaShop's whole structure ?

Thanks!

**Comment by ChazyTheBest** [comment]
Hi, I've updated the request, reworded it and added some more info. Hopefully it is clear and has enough information to present an interesting improvement request.

**Comment by matks** [comment]
Hi, sorry for late answer. It looks like this is the same topic / project that https://github.com/PrestaShop/PrestaShop/issues/20333

If we have a single `public/` folder, and only this folder is exposed to the web, everything is solved if I am not wrong?

@ChazyTheBest is it OK if I close this issue as duplicate topic of https://github.com/PrestaShop/PrestaShop/issues/20333

**Comment by ChazyTheBest** [comment]
As long as it is also considered to replace `mod_rewrite` ([which shouldn't be used for this in the first place](https://httpd.apache.org/docs/current/rewrite/avoid.html)) with `FallbackResource`. Unless I'm missing some use case which is not covered by it, I think it'd be a good idea to follow their recommendations.

**Comment by matks** [comment]
@PierreRambaud I am no expert in httpd config. Can you provide your knowledge about `mod_rewrite` usage?
